### ISSUE TITLE
Upgrade and check the custom plugin documentation

### ISSUE THREAD

**Issue by diddi-** [PyCQA/pylint#5922]
### Question

The documentation for How to write a Pylint plugin uses a function `get_checker`, but it's not described where this function can be found/imported from.

https://pylint.pycqa.org/en/latest/how_tos/plugins.html
```
...
def load_configuration(linter):

  name_checker = get_checker(linter, NameChecker)
...
```

### Documentation for future user

I'd expect the example on https://pylint.pycqa.org/en/latest/how_tos/plugins.html to include relevant import statements for the `get_checker()` function and also possible references to other modules/libraries if needed.

I can't find the function anywhere so can't suggest what should be added to the docs.

**Comment by DanielNoord** [comment]
I wasn't around when that documentation was written, so I don't know what happened to the method. I can help you with writing your plugin, so if there is any other issues please let me know!

We will take a look at the plugin docs and update them accordingly soon(™).

**Comment by diddi-** [comment]
Thanks. For the plugin I'm writing now I don't need any configuration so it's not an issue for me at the moment. I was just experimenting and curious what the function is and does.

Might get back with more specific questions when/if I need that functionality :)

**Comment by elbakramer** [comment]
This might work?

```python
def get_checker(linter: PyLinter, cls: Type[BaseChecker]) -> Optional[BaseChecker]:
    checkers = linter.get_checkers()
    checkers = filter(lambda checker: isinstance(checker, cls), checkers)
    checker = next(checkers, None)
    return checker
```

**Comment by DanielNoord** [comment]
Yeah, although I don't think we have any need for that function ourselves. I think it's better to just fix the documentation and remove the reference to a removed function.
